actors/miner/src/lib.rs

```diff
@@ -1552,10 +1479,20 @@ impl Actor {
                 ));
             }
 
-            if let Some(ref commd) = precommit.unsealed_cid.as_ref().and_then(|c| c.0) {
-                if !is_unsealed_sector(commd) {
-                    return Err(actor_error!(illegal_argument, "unsealed CID had wrong prefix"));
+            if let Some(compact_commd) = &precommit.unsealed_cid {
+                if let Some(commd) = compact_commd.0 {
+                    if !is_unsealed_sector(&commd) {
+                        return Err(actor_error!(
+                            illegal_argument,
+                            "unsealed CID had wrong prefix"
+                        ));
+                    }
                 }
+            } else {
+                return Err(actor_error!(
+                    illegal_argument,
+                    "unspecified CompactCommD not allowed past nv21, need explicit None value for CC or CommD"
+                ));
```

Comment:
```rust
            match &precommit.unsealed_cid {
                Some(Some(compact_commd)) if !is_unsealed_sector(&commd) => {
                    return Err(actor_error!(illegal_argument, "unsealed CID had wrong prefix"));
                }
                None => return Err(actor_error!(
                                illegal_argument,
                                "unspecified CompactCommD not allowed past nv21, need explicit None value for CC or CommD")),
                _ => {},
            }
```

Reply:
(slightly easier to read, IMO)

Reply:
but won't work because there's a tuple struct in the middle, maybe this?

```rust
            match precommit.unsealed_cid.as_ref().and_then(|c| c.0.as_ref()) {
                Some(compact_commd) if !is_unsealed_sector(&compact_commd) => {
```

Reply:
Ah. It can also be written as:

```rust
            match &precommit.unsealed_cid {
                Some(CompactCommD(Some(commd))) if !is_unsealed_sector(commd) => {
                    return Err(actor_error!(illegal_argument, "unsealed CID had wrong prefix"));
                }
                None => return Err(actor_error!(
                                illegal_argument,
                                "unspecified CompactCommD not allowed past nv21, need explicit None value for CC or CommD")),
                _ => {},
            }
```